packages/grid/x-data-grid/src/lib/createDetectElementResize/index.js

```diff
@@ -93,7 +93,7 @@ export default function createDetectElementResize(nonce, hostWindow) {
     ' { from { opacity: 0; } to { opacity: 0; } } ';
   var animationStyle = keyframeprefix + 'animation: 1ms ' + animationName + '; ';
 
-  var createStyles = function createStyles(doc) {
+  var createStyles = function createStyles(doc, root) {
     if (!doc.getElementById('muiDetectElementResize')) {
```

Comment:
```suggestion
    if (!root.getElementById('muiDetectElementResize')) {
```

Otherwise, the resize won't work if the grid was rendered in a Shadow DOM when there was another grid outside of the Shadow DOM